pkg/httpclient/client_test.go

```diff
@@ -0,0 +1,144 @@
+package httpclient
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNew(t *testing.T) {
+	acsToken := "acsToken"
+	url := "https://dcos.io"
+
+	conf := config.New()
+	conf.SetURL("https://dcos.io")
+	conf.SetACSToken(acsToken)
+
+	client := New(conf)
+
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: conf.TLS().Insecure,
+		RootCAs:            conf.TLS().RootCAs,
+	}
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	backendClient := &http.Client{
+		Transport: transport,
+		Timeout:   conf.Timeout(),
+	}
+	expectedClient := &Client{
+		acsToken: acsToken,
+		baseURL:  url,
+		client:   backendClient,
+	}
+
+	require.Equal(t, expectedClient, client)
+}
+
+func TestGet(t *testing.T) {
+	body := `{"cluster":"DC/OS"}`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			t.Errorf("Expected `GET` request, got `%s`", r.Method)
+		}
+		if r.URL.EscapedPath() != "/get" {
+			t.Errorf("expected request to `/get, got `%s`", r.URL.EscapedPath())
+		}
+
+		w.Write([]byte(body))
+	}))
+	defer ts.Close()
+
+	conf := config.New()
+	conf.SetURL(ts.URL)
+
+	client := New(conf)
+
+	resp, err := client.Get("/get")
+	require.NoError(t, err)
+
+	respBody, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Equal(t, body, string(respBody))
+}
+
+func TestPost(t *testing.T) {
+	type bodyStruct struct {
+		Cluster string
+	}
+	body := `{"cluster":"DC/OS"}`
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Errorf("Expected `POST` request, got `%s`", r.Method)
+		}
```

Comment:
All 3 liners like this in test http handlers can become one-liners, eg. `assert.Equal(t, "POST", r.Method)`